### PR TITLE
Fs/http sniffers

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

I've added the following snippet to the blog-api demo InfoController:
```php
file_get_contents(__FILE__);
file_put_contents(dirname(__DIR__).'/runtime/1.txt', 'smth');
$fp = fopen("example_homepage.txt", "w");
fwrite($fp, 'smth');
fclose($fp);
```
And got the following:

#### The index additional info:
```json
"file": {
    "read": 2,
    "write": 2,
    "mkdir": 1
},
```
#### The common result addition info:
```json
"Yiisoft\\Yii\\Debug\\Collector\\FileStreamCollector": {
    "read": [
        {
            "path": "\/Users\/xepozz\/PhpstormProjects\/yiisoft\/yii-dev-tool\/dev\/demo\/blog-api\/src\/InfoController.php",
            "args": []
        },
        {
            "path": "\/Users\/xepozz\/PhpstormProjects\/yiisoft\/yii-dev-tool\/dev\/demo\/blog-api\/src\/InfoController.php",
            "args": []
        }
    ],
    "write": [
        {
            "path": "\/Users\/xepozz\/PhpstormProjects\/yiisoft\/yii-dev-tool\/dev\/demo\/blog-api\/runtime\/1.txt",
            "args": []
        },
        {
            "path": "\/Users\/xepozz\/PhpstormProjects\/yiisoft\/yii-dev-tool\/dev\/demo\/blog-api\/example_homepage.txt",
            "args": []
        }
    ],
    "mkdir": [
        {
            "path": "\/Users\/xepozz\/PhpstormProjects\/yiisoft\/yii-dev-tool\/dev\/demo\/blog-api\/runtime\/debug\/2023-02-04\/63dea2b0c83132.65824852",
            "args": {
                "mode": 509,
                "options": 9
            }
        }
    ]
},
```

Rewrote tests because FS accessing affect tests.

Performance can be enhanced

### CODE REVIEW

File: config/common.php [hunk 1]
```diff
@@ -51,4 +55,19 @@
             $logLevel
         );
     },
+    FilesystemStreamCollector::class => [
+        '__construct()' => [
+            'ignoredPathPatterns' => [
+                //'/ClosureExporter/',
```
Comment (samdark) [inline config/common.php:61]: Do we need this commented code?

File: src/Collector/HttpStreamCollector.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    public function getCollected(): array
+    {
+        return $this->requests;
+    }
+
+    public function startup(): void
+    {
+        $this->isActive = true;
+        HttpStreamProxy::register();
+        // TODO: add cURL support, maybe through proxy?
```
Comment (samdark) [inline src/Collector/HttpStreamCollector.php:26]: Do you plan to implement it in this PR? If you plan it for later, please create an issue about it.

**Review by samdark — APPROVED** [review]

File: src/Collector/HttpStreamCollector.php (hunk 2 above)
Comment (xepozz) [reply src/Collector/HttpStreamCollector.php:26]: Unfortunately, It's impossible to do it now because PHP doesn't support it

Comment (xepozz) [reply src/Collector/HttpStreamCollector.php:26]: https://github.com/yiisoft/yii-debug/issues/192

File: config/common.php (hunk 1 above)
Comment (xepozz) [reply config/common.php]: Moved to property description

**Review by samdark — APPROVED** [review]

**Pull request merged**